py-polars/polars/functions/lazy.py

```diff
@@ -87,7 +88,11 @@
 
 
 def col(
-    name: str | PolarsDataType | Iterable[str] | Iterable[PolarsDataType],
+    name: str
+    | PolarsDataType
+    | Iterable[str]
+    | Iterable[PolarsDataType]
+    | Iterable[str | PolarsDataType],
     *more_names: str | PolarsDataType,
 ) -> Expr:
```

Comment:
We could change return type here from `Expr` to `Expr | Iterable[Expr]` to solve [the current mypy error](https://github.com/pola-rs/polars/actions/runs/4843378073/jobs/8630947570?pr=8589), but this results in other errors as below.

```
❯ make pre-commit
.venv/bin/black .
All done! ✨ 🍰 ✨
194 files left unchanged.
.venv/bin/blackdoc .

All done! ✨ 🍰 ✨
259 files left unchanged.
.venv/bin/ruff .
.venv/bin/typos ..
.venv/bin/python scripts/check_stacklevels.py
cargo fmt --all
dprint fmt
.venv/bin/mypy
polars/utils/various.py:273: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "str"  [union-attr]
polars/utils/various.py:284: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "str"  [union-attr]
polars/utils/various.py:284: note: Error code "union-attr" not covered by "type: ignore" comment
polars/utils/various.py:287: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "str"  [union-attr]
polars/utils/various.py:288: error: Unsupported left operand type for + ("Iterable[Expr]")  [operator]
polars/utils/various.py:288: note: Left operand is of type "Union[Expr, Iterable[Expr]]"
polars/utils/various.py:289: error: Unsupported left operand type for + ("Iterable[Expr]")  [operator]
polars/utils/various.py:289: note: Left operand is of type "Union[Expr, Iterable[Expr]]"
polars/utils/various.py:295: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "apply"  [union-attr]
polars/utils/various.py:301: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "map_dict"  [union-attr]
polars/utils/various.py:305: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "cast"  [union-attr]
polars/utils/_parse_expr_input.py:84: error: Item "Series" of "Union[Expr, WhenThen, WhenThenThen, Series, Iterable[Expr]]" has no attribute "meta"  [union-attr]
polars/utils/_parse_expr_input.py:84: error: Item "Iterable[Expr]" of "Union[Expr, WhenThen, WhenThenThen, Series, Iterable[Expr]]" has no attribute "meta"  [union-attr]
polars/utils/_parse_expr_input.py:86: error: Argument 1 to "_expr_output_name" has incompatible type "Union[Expr, WhenThen, WhenThenThen, Series, Iterable[Expr]]"; expected "Expr"  [arg-type]
polars/utils/_parse_expr_input.py:90: error: Incompatible return value type (got "Union[Expr, WhenThen, WhenThenThen, Series, Iterable[Expr], Any]", expected "Expr")  [return-value]
polars/utils/_parse_expr_input.py:90: error: Item "Iterable[Expr]" of "Union[Expr, WhenThen, WhenThenThen, Series, Iterable[Expr]]" has no attribute "alias"  [union-attr]
polars/io/excel/_write_utils.py:325: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "map"  [union-attr]
polars/utils/_construction.py:576: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "cast"  [union-attr]
polars/utils/_construction.py:578: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "cast"  [union-attr]
polars/utils/_construction.py:580: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "cast"  [union-attr]
polars/utils/_construction.py:587: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "_pyexpr"  [union-attr]
polars/series/list.py:440: error: Item "Iterable[Expr]" of "Union[Expr, Iterable[Expr]]" has no attribute "arr"  [union-attr]
```
This is because other functions need to be modified as well.